File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int sell=0,buy=INT_MIN;
        
        for(int &i:prices){
            int temp=buy;
            buy=max(buy,sell-i);
            sell=max(sell,buy+i);
        }
        
        return sell;
    }
};

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit=0;
        int buyPrice=prices[0];
        
        for(int currPrice:prices){
            if(currPrice>buyPrice){
                profit+=(currPrice-buyPrice);
            }
            buyPrice=currPrice;
        }
        
        return profit;
    }
}